src/components/GlobalPreferences/Notifications/Notifications.js

```diff
@@ -1,24 +1,178 @@
-import React from 'react'
-import { Box } from '@aragon/ui'
-
-function Notifications() {
-  return (
-    <Box heading="Email notifications">
-      <label>
-        App notifications and reminders (upcoming votes, created incoming or
-        outgoing transactions, added or removed tokens, etc.).{' '}
-        <input type="checkbox" />
-      </label>
-      <label>
-        New releases and features announcements. <input type="checkbox" />
-      </label>
-      <label>
-        Calendar reminders of selected events or tasks.{' '}
-        <input type="checkbox" />
-      </label>
-      <button>Manage triggers</button>
-    </Box>
+import React, { useState, useEffect, useCallback } from 'react'
+import PropTypes from 'prop-types'
+import { AppType } from '../../../prop-types'
+import { ButtonText, LoadingRing, useTheme } from '@aragon/ui'
+import ManageNotifications from './ManageNotifications'
+import NotificationsLogin from './NotificationsLogin'
+import {
+  NotificationsVerify,
+  NotificationsPreVerify,
+} from './NotificationsVerify'
+import NotificationsVerifyBox from './NotificationsVerifyBox'
+import { isAuthTokenValid } from './notification-service-api'
+
+import {
+  AUTH_UNAUTHENTICATED,
+  AUTH_PREVERIFY,
+  AUTH_AUTHENTICATED,
+  AUTH_AUTHENTICATION_FAILED,
+  NOTIFICATION_SERVICE_EMAIL_KEY,
+  NOTIFICATION_SERVICE_TOKEN_KEY,
+  VERIFY_SUBSECTION,
+  AUTH_AUTHENTICATING,
+} from './constants'
+
+// Hook responsible for deriving the authState from localStorage values and
+// providing setters which update the localStorage
+function useAuthState() {
+  const [authState, setAuthState] = useState(AUTH_UNAUTHENTICATED)
+  const [email, setEmail] = useState(
+    localStorage.getItem(NOTIFICATION_SERVICE_EMAIL_KEY)
+  )
+  const [token, setToken] = useState(
+    localStorage.getItem(NOTIFICATION_SERVICE_TOKEN_KEY)
+  )
+
+  useEffect(() => {
+    if (!email && !token) {
+      setAuthState(AUTH_UNAUTHENTICATED)
+      return
+    }
+
+    if (email && !token) {
+      setAuthState(AUTH_PREVERIFY)
+      return
+    }
+
+    if (email && token) {
```

Comment:
At this point we can assume this are both `truthy` no? No need for this check

Reply:
It can't happen but for someone manipulating local storage. Do you think we should handle it?

Reply:
I wouldn't worry about it happening, but because it could happen either have a last "default" `return null` or handling it specifically (last return is easiest, and if someone meddles with `localStorage` they should know better)